testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -0,0 +1,621 @@
+    fn mine_block(&mut self) -> Option<NakamotoBlock> {
+        debug!("block miner thread ID is {:?}", thread::current().id());
+        neon_node::fault_injection_long_tenure();
+
+        let burn_db_path = self.config.get_burn_db_file_path();
+
+        // NOTE: read-write access is needed in order to be able to query the recipient set.
+        // This is an artifact of the way the MARF is built (see #1449)
+        let mut burn_db =
+            SortitionDB::open(&burn_db_path, true, self.burnchain.pox_constants.clone())
+                .expect("FATAL: could not open sortition DB");
+
+        let mut chain_state = neon_node::open_chainstate_with_faults(&self.config)
+            .expect("FATAL: could not open chainstate DB");
+
+        let mut mem_pool = self
+            .config
+            .connect_mempool_db()
+            .expect("Database failure opening mempool");
+
+        let target_epoch_id =
+            SortitionDB::get_stacks_epoch(burn_db.conn(), self.burn_block.block_height + 1)
+                .ok()?
+                .expect("FATAL: no epoch defined")
+                .epoch_id;
+        let mut parent_block_info = self.load_block_parent_info(&mut burn_db, &mut chain_state)?;
+        let vrf_proof = self.make_vrf_proof()?;
+
+        if self.last_mined_blocks.is_empty() {
+            if parent_block_info.parent_tenure.is_none() {
+                warn!(
+                    "Miner should be starting a new tenure, but failed to load parent tenure info"
+                );
+                return None;
+            }
+        }
+
+        // create our coinbase if this is the first block we've mined this tenure
+        let tenure_start_info = if let Some(ref par_tenure_info) = parent_block_info.parent_tenure {
+            let parent_block_id = parent_block_info.stacks_parent_header.index_block_hash();
+            let current_miner_nonce = parent_block_info.coinbase_nonce;
+            let tenure_change_tx = self.generate_tenure_change_tx(
+                current_miner_nonce,
+                parent_block_id,
+                par_tenure_info.parent_tenure_blocks,
+                self.keychain.get_nakamoto_pkh(),
+            )?;
+            let coinbase_tx = self.generate_coinbase_tx(
+                current_miner_nonce + 1,
+                target_epoch_id,
+                vrf_proof.clone(),
+            );
+            Some(NakamotoTenureStart {
+                coinbase_tx,
+                // TODO (refactor): the nakamoto block builder doesn't use this VRF proof,
+                //  it has to be included in the coinbase tx, which is an arg to the builder.
+                //  we should probably just remove this from the nakamoto block builder.
+                vrf_proof: vrf_proof.clone(),
+                tenure_change_tx,
+            })
+        } else {
+            None
+        };
+
+        parent_block_info.stacks_parent_header.microblock_tail = None;
```

Comment:
Might also want to set the seq to 0

Reply:
The sequence number is part of that `StacksMicroblockHeader` struct, which is being set to `None` here, right?